Port to Python 3

kiy4h opened the pull request.

Commit:
Port to Python 3

Commit:
set pep8

kiy4h committed:
Port to Python 3

dictdlib.py: add cmp function in sortfunc()
wordsactivity.py: convert translations and definitions to str from bytes using .decode()

kiy4h committed:
Merge branch 'python3' into python3

tchx84 commented:
I tested this, simply wrote "table" on the word field to translate it from English to Spanish and got the following error:

```
Traceback (most recent call last):
  File "/home/sugar/words-activity/wordsactivity.py", line 615, in __totranslate_activated_cb
    self._translate()
  File "/home/sugar/words-activity/wordsactivity.py", line 641, in _translate
    translations[i] = translations[i].decode()
AttributeError: 'str' object has no attribute 'decode'
```

tchx84 commented on dictdlib.py:
```diff
@@ -59,6 +59,7 @@ def b64_decode(str):
         shiftval += 6
     return retval
 
+
```
A typo?

kiy4h replied:
Whoops, fixed.

tchx84 reviewed (changes requested):
Please test the activity as well.

kiy4h committed:
fix a few things
wordsactivity.py: remove .decode() since open() was fixed
dictdlib.py: undo accidental newline change

kiy4h commented:
I forget to test after f3f2527, it won't happen again. It was now tested in a VM and running smoothly.

quozl commented:
Thanks.  Great progress.  Tests out fine for me.

However, you should have opened this pull request to merge into the _master_ branch.  I've taken your changes relative to @pro-panda's and pushed to _master_ along with my own changes.

kiy4h commented:
Whoops, sorry!

Kiy4h commented:
Whoops, sorry!

Kiy4h commented:
Whoops, sorry!

Kiy4h commented:
Whoops, sorry!

quozl commented:
No worries, I can git.  :grin:
